lib/most_frequent_integer.rb

```diff
@@ -3,5 +3,52 @@
 # exists, return nil.
 
 def find_most_frequent_integer(arr)
-  # put your code here
+
+sortarr = arr.sort
+indexarr = Array.new
+n = 0
+
+if sortarr.empty?
+    return nil
+else
+    i = sortarr[0]
+end
+
+while n < sortarr.length && i <= sortarr.last
+
+    indexarr[n] = sortarr.each_index.select{|x| sortarr[x] == i} 
+    i = i + 1
+    n = n + 1
+    
 end
+
+m = 0
```

Comment:
Having two iterative variables, `m` and `p` that have the same starting value and are effectively being used in the same manner seems redundant.